Sources/Utilities/ALMessage+Extension.swift

```diff
@@ -205,6 +205,17 @@ extension ALMessage: ALKChatViewModelProtocol {
     public var messageMetadata: NSMutableDictionary? {
         return metadata
     }
+    
+    public var platformSource: String? {
+        guard let sourceChannel = ALChannelDBService().getChannelByKey(channelKey) else { return ""}
+        if sourceChannel.platformSource == nil {
+            guard let channel = ALChannelService().getChannelByKey(channelKey), let sourceFromMeta = channel.metadata.value(forKey: "source") else {return nil}
+            ALChannelDBService().updatePlatformSource(channelKey, platformSource: sourceFromMeta as? String)
+        }
```

Comment:
you can move this if check previous line

Reply:
okay I will do this

Reply:
@Sathyan-KM actually I checked the sourceChannel is used outside the if block also and the if block is only need to be executed when there is no data inside the platformSource. So removing guard let will not work in this case.

Reply:
moved the guard let channel inside the if block to reduce the use of guard.